src/components/ble/weather/WeatherService.h

```diff
@@ -146,6 +146,9 @@ namespace Pinetime {
       static bool CompareTimelineEvents(const std::unique_ptr<WeatherData::TimelineHeader>& first,
                                         const std::unique_ptr<WeatherData::TimelineHeader>& second);
 
+      template <class T>
+      std::unique_ptr<T>& GetEventOfType(WeatherData::eventtype eventType);
```

Comment:
Templates... this probably adds a version of this function of each event of type `eventtype`. Which is why I kinda just added the few that might be immediately needed.

It might be better to deprecate the named functions and force everything to use this one, to avoid at least one level of duplication.

Reply:
Nope, the template is only depedant on the `class T` type, and it will only create a version for each one of those. I purposely did not make the event type a template parameter. I checked the final binary. The compiler is even smart enough to inline the call to this function for each instance.

Reply:
> and it will only create a version for each one of those

Yep.

> The compiler is even smart enough to inline the call to this function for each instance.

That's nice. Though without LTO the unused ones will remain right now, right?